### PR TITLE
Enable cudagraphs in cpp wrapper

### LINKED ISSUE #130878: cpp wrapper doesn't work with cudagraphs
Remove `"triton.cudagraphs": False` from [here](https://github.com/pytorch/pytorch/blob/7919f0b952272b83886021b172a8d56a893917ba/torch/_inductor/compile_fx.py#L1314C17-L1314C44), and then run the following with `TORCHINDUCTOR_ABI_COMPATIBLE=1 TORCHINDUCTOR_CPP_WRAPPER=1`

```
import torch

def f(x):
    return torch.sin(x)

optimized = torch.compile(f, mode="reduce-overhead")
print(optimized(torch.randn(10, device="cuda")))
```

, which gives the following error,
```
raceback (most recent call last):
  File "/data/users/binbao/pytorch/test2.py", line 8, in <module>
    print(optimized(torch.randn(10, device="cuda")))
  File "/data/users/binbao/pytorch/torch/_dynamo/eval_frame.py", line 435, in _fn
    return fn(*args, **kwargs)
  File "/data/users/binbao/pytorch/test2.py", line 3, in f
    def f(x):
  File "/data/users/binbao/pytorch/torch/_dynamo/eval_frame.py", line 602, in _fn
    return fn(*args, **kwargs)
  File "/data/users/binbao/pytorch/torch/_functorch/aot_autograd.py", line 1024, in forward
    return compiled_fn(full_args)
  File "/data/users/binbao/pytorch/torch/_functorch/_aot_autograd/runtime_wrappers.py", line 321, in runtime_wrapper
    all_outs = call_func_at_runtime_with_args(
  File "/data/users/binbao/pytorch/torch/_functorch/_aot_autograd/utils.py", line 120, in call_func_at_runtime_with_args
    out = normalize_as_list(f(args))
  File "/data/users/binbao/pytorch/torch/_functorch/_aot_autograd/runtime_wrappers.py", line 500, in wrapper
    return compiled_fn(runtime_args)
  File "/data/users/binbao/pytorch/torch/_inductor/codecache.py", line 1199, in __call__
    return self.current_callable(inputs)
  File "/data/users/binbao/pytorch/torch/_inductor/compile_fx.py", line 1028, in run
    return compiled_fn(new_inputs)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 373, in deferred_cudagraphify
    fn, out = cudagraphify(model, inputs, new_static_input_idxs, *args, **kwargs)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 403, in cudagraphify
    return manager.add_function(
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 2093, in add_function
    return fn, fn(inputs)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 1845, in run
    out = self._run(new_inputs, function_id)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 1936, in _run
    return self.run_eager(new_inputs, function_id)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 2059, in run_eager
    return node.run(new_inputs)
  File "/data/users/binbao/pytorch/torch/_inductor/cudagraph_trees.py", line 626, in run
    assert len(new_inputs) == 0
AssertionError
```

cc @ezyang @anijain2305 @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames

### PR DESCRIPTION
**Pull request by exclamaforte**
Fixes https://github.com/pytorch/pytorch/issues/130878

Summary: Enables cudagraphs in cpp wrapper by clearing inputs.

Generated, non-cpp wrapper code:
```python
def call(args):
    arg0_1, = args
    args.clear()
    assert_size_stride(arg0_1, (10, ), (1, ))
    with torch.cuda._DeviceGuard(0):
        torch.cuda.set_device(0)
        buf0 = empty_strided_cuda((10, ), (1, ), torch.float32)
        # Topologically Sorted Source Nodes: [sin], Original ATen: [aten.sin]
        stream0 = get_raw_stream(0)
        triton_poi_fused_sin_0.run(arg0_1, buf0, 10, grid=grid(10), stream=stream0)
        del arg0_1
    return (buf0, )
```
vs generated cpp wrapper code:
```python
def _wrap_func(f):
    def g(args):
        input_tensors = [arg if isinstance(arg, torch.Tensor) else torch.tensor(arg) for arg in args]
        input_handles = torch._C._aoti.unsafe_alloc_void_ptrs_from_tensors(input_tensors)
        # new:
        args.clear()
        # end new

        output_handles = f(input_handles)
        output_tensors = torch._C._aoti.alloc_tensors_by_stealing_from_void_ptrs(output_handles)
        return output_tensors

    return g

```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: .ci/pytorch/test.sh
Comment (eellison) [inline .ci/pytorch/test.sh:476]: I think this is probably unintended, but if it's not, should go in as a separate change

File: test/inductor/test_cudagraph_trees.py
Comment (eellison) [inline test/inductor/test_cudagraph_trees.py:2454]: Use `run_twc`? and check `assertEqual` to eager results ? also, should not print in test.

File: torch/_inductor/codegen/cpp_wrapper_cpu.py
Comment (eellison) [inline torch/_inductor/codegen/cpp_wrapper_cpu.py:1158]:
```
for inp in args:
   del inp
```

Is a no-op. 

Just need `args.clear()`

File: torch/_inductor/compile_fx.py
Comment (eellison) [inline torch/_inductor/compile_fx.py:1272]: I dont think we want to patch this to True or False, just leave it to what was previously configured.

**Review by eellison — COMMENTED** [review]
a few comments

File: .ci/pytorch/test.sh
Comment (exclamaforte) [reply .ci/pytorch/test.sh:476]: Yes correct, I just added this to benchmark the change on the perf dashboard. It will not go in, was planning on removing it after removing the WIP.

File: torch/_inductor/compile_fx.py
Comment (exclamaforte) [reply torch/_inductor/compile_fx.py:1272]: Sorry, this is another that I was planning on removing with the WIP, was just there to test in the perf dashboard. I'll add a `# TODO remove before merging` next time.

**Review by eellison — APPROVED** [review]
looks good !

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**
